macros/get_query_comment.sql

```diff
@@ -0,0 +1,65 @@
+{# have to reimplement is_incremental because node.type hasn't populated #}
+{% macro custom_is_incremental(node) %}
+    {{ return(
+        node.config.materialized == 'incremental'
+        and not dbt_snowflake_monitoring.custom_should_full_refresh(node)
+    ) }}
+{% endmacro %}
+
+{% macro custom_should_full_refresh(node) %}
+  {% set config_full_refresh = node.config.get('full_refresh') %}
+  {% if config_full_refresh is none %}
+    {% set config_full_refresh = flags.FULL_REFRESH %}
+  {% endif %}
+  {% do return(config_full_refresh) %}
+{% endmacro %}
+
+{% macro get_query_comment(node) %}
+{%- set comment_dict = {} -%}
+{%- do comment_dict.update(
+    app='dbt',
+    dbt_version=dbt_version,
+    target_name=target.name,
+    target_database=target.database,
+    target_schema=target.schema,
+    invocation_id=invocation_id,
+) -%}
+{%- if env_var('DBT_CLOUD_PROJECT_ID', False) -%}
```

Comment:
We can add the capability for users to add their own environment variables/other stuff they might want to slice on.